tests/unit/httpd_test.cc

```diff
@@ -2019,14 +2019,20 @@ future<> test_client_close_connection(bool chunked) {
                         });
                     },
                     http::reply::status_type::ok);
+              if (!chunked || (_ == 1)) {
                 BOOST_REQUIRE_NO_THROW(make_request.get());
+              } else {
+                // FIXME chunked encoded bodies are not yet checked by client to be
+                // under-read, so 2nd request will receive the tail of the 1st response
+                BOOST_REQUIRE_THROW(make_request.get(), httpd::response_parsing_exception);
```

Comment:
I don't like to have tests which enshrine **wrong** behavior. I'd much rather have a test that should pass, and for now we know fails (i.e., "xfail" in pytest lingo). I don't know how to do this in unit tests.
But it still doesn't mean we can have a test which insists that a known bug can't be fixed...